[강진호] week7

jinhok96 opened the pull request:
## 요구사항

### 기본

- [x] 상단 네비게이션 바, 푸터가 랜딩 페이지와 동일한 스타일과 규칙으로 만들어졌나요? (week 1 ~ 3 요구사항 참고)

- [x] 상단 네비게이션 바에 프로필 영역의 데이터는 https://bootcamp-api.codeit.kr/docs 에 명세된 “/api/sample/user”를 활용했나요?

- [x] 상단 네비게이션 바에 프로필 영역의 데이터가 없는 경우 “로그인” 버튼이 보이도록 만들었나요?

- [x] 폴더 소유자, 폴더 이름 영역, 링크들에 대한 데이터는 “/api/sample/folder”를 활용했나요?

- [x] Static, no image, Hover 상태 디자인을 보여주는 카드 컴포넌트를 만들었나요?

- [x] Hover 상태에서 이미지가 기본 크기의 130%로 커지나요?

- [x] 카드 컴포넌트를 클릭하면 해당 링크로 새로운 창을 띄워서 이동하나요?

- [x] Tablet에서 카드 리스트가 화면의 너비 1124px를 기준으로 이상일 때는 3열로 작을 때는 2열로 배치되나요?

- [x] Tablet, Mobile에서 좌우 최소 여백은 32px 인가요?

### 심화

- [ ] 커스텀 hook을 만들어 활용했나요?

## 주요 변경사항

- Card 컴포넌트 hover일 때 ease-in-out 효과 추가했습니다.
- 프로젝트 규모가 크지 않아 Header/Main/Footer/Common으로 컴포넌트 구조를 나눴습니다.
- 커스텀 훅은 사용할 부분을 찾지 못해 추가하지 않았습니다.

## 스크린샷

https://jinhok96-linkbrary.netlify.app/

## 멘토에게

- 새로 추가하는 부분이 많아 한 커밋 안에 변경사항이 많습니다. 앞으로 기능을 추가하거나 수정할 경우 커밋을 더 작은 단위로 올릴 예정입니다.
- 셀프 코드 리뷰를 통해 질문 이어가겠습니다.

jinhok96 committed:
refactor: 리액트 초기 파일 구조 정리 및 테스트
	modified:   .gitignore
	new file:   README.old.md
	deleted:    public/favicon.ico
	modified:   public/index.html
	deleted:    public/logo192.png
	deleted:    public/logo512.png
	deleted:    public/manifest.json
	deleted:    public/robots.txt
	modified:   src/App.css
	modified:   src/App.js
	deleted:    src/App.test.js
	modified:   src/index.js
	deleted:    src/logo.svg
	new file:   src/pages/Shared/Shared.css
	new file:   src/pages/Shared/Shared.js
	deleted:    src/reportWebVitals.js
	deleted:    src/setupTests.js

jinhok96 committed:
test: Gnb.js 생성
	modified:   src/App.js
	new file:   src/components/Gnb.css
	new file:   src/components/Gnb.js

jinhok96 committed:
chore: css reset 적용
	new file:   src/assets/styles/reset.css
	modified:   src/index.css

jinhok96 committed:
chore: 기본 폰트로 pretendard 적용
	modified:   public/index.html
	modified:   src/index.css

jinhok96 committed:
chore: 1rem=10px 적용
	modified:   src/assets/styles/reset.css

jinhok96 committed:
chore: 공통으로 사용되는 여러 스타일시트 추가
	new file:   src/assets/styles/colors.css
	new file:   src/assets/styles/common.css
	new file:   src/assets/styles/utility.css
	modified:   src/index.css

jinhok96 committed:
chore: css reset에 margin, padding 0으로 초기화
	modified:   src/assets/styles/reset.css

jinhok96 committed:
chore: 중복된 스타일 제거 및 폰트 기본값 업데이트
	modified:   src/index.css

jinhok96 committed:
chore: Header 컴포넌트 추가
	new file:   src/components/Header.css
	new file:   src/components/Header.js

jinhok96 committed:
feat: 컴포넌트 외형 구현
- 링크, api, 네트워크 관련 등 제외
	modified:   src/App.js
	new file:   src/assets/images/akar-icons_facebook-fill.svg
	new file:   src/assets/images/akar-icons_twitter-fill.svg
	new file:   src/assets/images/akar-icons_youtube-fill.svg
	new file:   src/assets/images/ant-design_instagram-filled.svg
	new file:   src/assets/images/card-1.png
	new file:   src/assets/images/card-2.png
	new file:   src/assets/images/card-3.png
	new file:   src/assets/images/card-4.png
	new file:   src/assets/images/card-5.png
	new file:   src/assets/images/card-6.png
	new file:   src/assets/images/card-7.png
	new file:   src/assets/images/card-8.png
	new file:   src/assets/images/card-9.png
	new file:   src/assets/images/logo.svg
	new file:   src/assets/images/profile-img.png
	new file:   src/assets/images/search-icon.svg
	new file:   src/assets/images/user-avatar.png
	modified:   src/assets/styles/colors.css
	modified:   src/assets/styles/utility.css
	new file:   src/components/Card.css
	new file:   src/components/Card.js
	new file:   src/components/CardList.css
	new file:   src/components/CardList.js
	new file:   src/components/Footer.css
	new file:   src/components/Footer.js
	modified:   src/components/Gnb.css
	modified:   src/components/Gnb.js
	modified:   src/components/Header.css
	modified:   src/components/Header.js
	new file:   src/components/Main.css
	new file:   src/components/Main.js
	new file:   src/components/SearchBar.css
	new file:   src/components/SearchBar.js
	new file:   src/components/SharedHeaderContent.css
	new file:   src/components/SharedHeaderContent.js
	modified:   src/index.css
	modified:   src/pages/Shared/Shared.js

jinhok96 committed:
refactor: components 하위 파일 구조 개선
	renamed:    src/components/Footer.css -> src/components/Footer/Footer.css
	renamed:    src/components/Footer.js -> src/components/Footer/Footer.js
	renamed:    src/components/Gnb.css -> src/components/Header/Gnb.css
	renamed:    src/components/Gnb.js -> src/components/Header/Gnb.js
	renamed:    src/components/Header.css -> src/components/Header/Header.css
	renamed:    src/components/Header.js -> src/components/Header/Header.js
	renamed:    src/components/SharedHeaderContent.css -> src/components/Header/SharedHeaderContent.css
	renamed:    src/components/SharedHeaderContent.js -> src/components/Header/SharedHeaderContent.js
	renamed:    src/components/Card.css -> src/components/Main/Card.css
	renamed:    src/components/Card.js -> src/components/Main/Card.js
	renamed:    src/components/CardList.css -> src/components/Main/CardList.css
	renamed:    src/components/CardList.js -> src/components/Main/CardList.js
	renamed:    src/components/Main.css -> src/components/Main/Main.css
	renamed:    src/components/Main.js -> src/components/Main/Main.js
	renamed:    src/components/SearchBar.css -> src/components/Main/SearchBar.css
	renamed:    src/components/SearchBar.js -> src/components/Main/SearchBar.js
	modified:   src/pages/Shared/Shared.js

jinhok96 committed:
chore: api.js 추가
	new file:   src/services/api.js

jinhok96 committed:
chore: 로그인 버튼 추가
	modified:   src/assets/styles/colors.css
	modified:   src/assets/styles/utility.css
	new file:   src/components/Common/Button.css
	new file:   src/components/Common/Button.js
	new file:   src/components/Common/LoginButton.css
	new file:   src/components/Common/LoginButton.js
	modified:   src/services/api.js

jinhok96 committed:
feat: gnb 유저 프로필 기능 추가
- api에서 정보를 성공적으로 가져오면 유저 프로필 출력
- 실패하면 로그인 버튼 출력
	modified:   src/components/Common/Button.css
	modified:   src/components/Header/Gnb.css
	modified:   src/components/Header/Gnb.js
	modified:   src/services/api.js

jinhok96 committed:
feat: 폴더 정보 출력 기능 추가
	modified:   src/components/Header/Gnb.js
	modified:   src/components/Header/SharedHeaderContent.js
	modified:   src/services/api.js

jinhok96 committed:
chore: userInfo 객체 속성을 변수에 저장하여 사용
	modified:   src/components/Header/Gnb.js

jinhok96 committed:
chore: api 참고용 주석 수정
	modified:   src/services/api.js

jinhok96 committed:
chore: 날짜 계산 함수 추가
	modified:   src/components/Main/Card.js
	new file:   src/utils/timeAgo.js

jinhok96 committed:
chore: Card 컴포지션에 여러 props 적용
	modified:   src/components/Main/Card.js
	new file:   src/utils/formatDate.js

jinhok96 committed:
feat: Card 컴포넌트에 api 적용
	modified:   src/components/Main/CardList.js

jinhok96 committed:
fix: Card 컴포넌트에 key prop 추가
	modified:   src/components/Main/CardList.js

jinhok96 committed:
chore: Card 컴포넌트 이미지 적용
	modified:   src/components/Main/Card.css
	modified:   src/components/Main/Card.js

jinhok96 committed:
feat: Card 컴포넌트 onClick 이벤트 추가
	modified:   src/components/Main/Card.js

jinhok96 committed:
feat: Card 컴포넌트 no-image 상태 추가
	new file:   src/assets/images/card-no-image.svg
	modified:   src/components/Main/Card.js

jinhok96 committed:
feat: Card 컴포넌트 hover 상태 추가
	modified:   src/components/Main/Card.css

jinhok96 committed:
refactor: Card background 관련 속성 하나로 병합
	modified:   src/components/Main/Card.js

jinhok96 committed:
chore: Card hover 상태에서 border 삭제
	modified:   src/components/Main/Card.css

jinhok96 committed:
chore: 데이터 로딩에 사용할 커스텀 hook 추가
	new file:   src/hooks/useFetchData.js

jinhok96 committed:
chore: 사용하지 않는 커스텀 훅 삭제
	deleted:    src/hooks/useFetchData.js

jinhok96 committed:
chore: 필요없는 handleLoad 제거
	modified:   src/components/Header/Gnb.js
	modified:   src/components/Header/SharedHeaderContent.js

jinhok96 committed:
chore: footer 아이콘 alt 추가
	modified:   src/components/Footer/Footer.js

jinhok96 committed:
fix: babel-preset-eract-app 관련 오류 메시지 해결
- @babel/plugin-proposal-private-property-in-object 설치
	modified:   package-lock.json
	modified:   package.json

SeolJaeHyeok commented on src/App.js:
```diff
@@ -1,23 +1,11 @@
-import logo from './logo.svg';
-import './App.css';
+// import { useEffect, useState } from "react-dom";
```
안쓰는 패키지는 주석 대신 지워줄게요! 
참고로 `react-dom` 패키지 내부에 `useEffect`, `useState`는 없습니다!!

SeolJaeHyeok commented on src/App.js:
```diff
@@ -1,23 +1,11 @@
-        </a>
-      </header>
+    <div>
+      <Shared></Shared>
```
저는 개인적으로 `Shared` 라는 이름이 조금 어색한거 같아요.  `Shared` 내부를 보면 아래와 같은 구조로 만들어주셨는데요.

```javascript
 <div>
      <Header />
      <Main />
      <Footer />
 </div>
```

일반적으로 Header, Main, Footer와 같은 화면 안의 정해진 공간을 차지하는 요소들은 `Layout`이라는 컴포넌트를 만들어서 관리합니다! 
추가적으로 Header와 Footer를 제외한 내부 컨텐츠들은 바뀔 수 있기 때문에 children props를 사용해서 레이아웃은 유지한 채 컨텐츠만 넘겨주는 방식으로 구현을 많이 합니다! 

참고 : [children](https://react.dev/learn/passing-props-to-a-component#passing-jsx-as-children)

++ 
`Shared`라는 이름은 공통으로 사용되는 컴포넌트(Ex. Button, Select, Input 등)에 붙여주는게 좋아보여요!

jinhok96 replied:
피그마에서 페이지 이름이 landing, siginin, signup, shared, ... 로 되어있어서 Shared로 설정하였습니다! 바뀔 수 있는 컨텐츠들은 children props 사용해보겠습니다!

SeolJaeHyeok commented on src/components/Common/LoginButton.js:
```diff
@@ -0,0 +1,9 @@
+import Button from "./Button";
+
+function LoginButton() {
+  const button = <Button className="login-button button-background-primary button-text-light" text="로그인" />;
+  return button;
```
지금은 일반 css 방식으로 스타일을 주입하고 있는데 이런 경우 className이 겹치면 스타일 또한 겹칠 수 있는 문제가 발생할 수 있습니다.
예를 들어, 코드 내에 `button-text-light`라는 className이 현재는 `LoginButton.css`에 선언이 되어 있지만 추가적으로 css 파일을 생성하고 이 파일 내부에 `button-text-light`를 선언하게 되면 스타일이 겹칠 수가 있어요

이런 문제를 해결하려고 나온게  [css module](https://www.tcpschool.com/react/react_styling_cssmodule)이어서 이 글 참고하시고 개발하시면 좋을거 같아요!

SeolJaeHyeok commented on src/components/Footer/Footer.js:
```diff
@@ -0,0 +1,25 @@
+        <p>FAQ</p>
+      </div>
+      <div className="social-link flex-row">
+        <img className="social-link-icon" src={facebook} alt="facebook"></img>
+        <img className="social-link-icon" src={twitter} alt="twitter"></img>
+        <img className="social-link-icon" src={youtube} alt="youtube"></img>
+        <img className="social-link-icon" src={instagram} alt="instagram"></img>
```
여기는 `map`으로 바꿔볼 수 있겠네요~

SeolJaeHyeok commented on src/services/api.js:
```diff
@@ -0,0 +1,136 @@
+export async function getUserInfo() {
+  const url = new URL("sample/user", API_URL);
+
+  try {
+    const response = await fetch(url);
+    const result = await response.json();
+    const errorMessage = result.error;
+
+    if (!response.ok) {
+      console.error("Response falied: ", errorMessage);
+      return null;
+    }
+
+    return result;
+  } catch (error) {
+    console.error("Fetch error: ", error);
+  }
+}
```
API 선언부에서 `try...catch`를 사용하여 에러를 캐치하고 로깅을 하는 것도 좋은데 지금은 API를 사용하는 컴포넌트에서도 `try...catch`를 사용하여 같은 로직을 수행하고 있어요.

이렇게 하는 것보다 API 선언부에서는 `catch`에 걸리는 error를 `throw`하고 API를 사용하는 부분의 `try...catch`에서 에러를 캐치하여 추가적인 처리(Ex. 에러 메시지 보여주기 등)을 해주는 것이 좋습니다!

SeolJaeHyeok commented on src/components/Header/Gnb.js:
```diff
@@ -0,0 +1,46 @@
+    try {
+      const result = await getUserInfo();
+      setUserInfo(result);
+    } catch (error) {
+      console.error(error);
+      setUserInfo(null);
+    }
```
API 호출부에서는 단순 로깅이 아니라 유저에게 에러를 보여줄 수 있는 처리를 해주는게 UX적으로 좋습니다!

++
userInfo의 초기값이 `null`로 설정되어 있고 요청이 실패하면 userInfo 값의 변화가 없기 때문에 `catch` 문 안에서 `setUserInfo(null)` 처리를 별도로 해줄 필요는 없습니다.

SeolJaeHyeok commented on src/components/Header/Gnb.js:
```diff
@@ -0,0 +1,46 @@
+    handleLoadUser();
+  }, []);
+
+  const userProfileImg = userInfo?.profileImageSource || "";
+  const userEmail = userInfo?.email || "";
```
👍

SeolJaeHyeok commented on src/components/Header/SharedHeaderContent.js:
```diff
@@ -0,0 +1,45 @@
+  const [folderInfo, setFolderInfo] = useState(null);
+
+  const handleLoadFolder = async () => {
+    try {
+      const result = await getFolderInfo();
+      setFolderInfo(result);
+    } catch (error) {
+      console.error(error);
+      setFolderInfo(null);
+    }
+  };
```
이 부분도 위에서 언급드린 것과 같이 수정하면 좋을거 같아요!

SeolJaeHyeok commented on src/components/Main/Card.js:
```diff
@@ -0,0 +1,30 @@
+    background: `#d9d9d9 url(${imageSource ? imageSource : NoImage}) no-repeat center / cover`,
+  };
+
+  const handleClick = () => {
+    window.open(url, "_blank", "noopener,noreferrer");
+  };
```
함수 이름은 되도록 구체적으로 작성하는게 좋습니다!

SeolJaeHyeok commented on src/components/Main/SearchBar.js:
```diff
@@ -0,0 +1,26 @@
+import searchIcon from "../../assets/images/search-icon.svg";
+
+function SearchBar() {
+  const [value, setValue] = useState("");
+
+  function handleChange(event) {
+    setValue(event.target.value);
+  }
```
여기도 조금 더 구체적인 네이밍이면 좋을거 같아요!

추가로 컴포넌트 내부의 함수 선언 방식이 화살표 함수와 함수 선언 방식이 혼재되어 있는데 통일하는게 좋습니다.
일반적으로 화살표 함수를 사용하시는걸 추천드려요!

SeolJaeHyeok commented on src/services/api.js:
```diff
@@ -0,0 +1,136 @@
+//   profileImageSource: "https://codeit-front.s3.ap-northeast-2.amazonaws.com/images/default_profile.png",
+// };
+export async function getUserInfo() {
+  const url = new URL("sample/user", API_URL);
+
```
API 사용하는 곳에서 URL을 생성하는 동일한 로직이 반복되고 있는데 이 부분 함수로 분리할 수 있을거 같죠!?!

SeolJaeHyeok reviewed (commented):
7주차 미션 고생많으셨습니다 진호님! 전반적을 잘 만들어주신 것 같아요👏👏👏
드리고 싶은 말씀은 아래 코멘트로 남겨놨으니 확인해주세요~

추가적으로  `ESLint`, `Prettier`까지 설정 해주시면 좋을거 같아요! 요즘 협업할 때는 ESLint, Prettier 설정은 필수이기 때문에 일반적으로 ESLint, Prettier는 프로젝트 시작 전에 설정을 하고 들어가는게 좋습니다.

설정 방법은 많이 알려져 있어서 찾아보신 다음 적용해주시면 될 거 같아요! 혹 어려움이 생기면 디스코드에 저 태그하신 다음 말해주세요~

jinhok96 committed:
modified:   .gitignore
	modified:   package-lock.json
	modified:   package.json

jinhok96 committed:
chore: eslint, prettier 적용
	modified:   .gitignore
	modified:   package-lock.json
	modified:   package.json
	modified:   src/App.js
	modified:   src/components/Common/Button.js
	modified:   src/components/Footer/Footer.js
	modified:   src/components/Header/Gnb.js
	modified:   src/components/Header/Header.js
	modified:   src/components/Header/SharedHeaderContent.js
	modified:   src/components/Main/Card.js
	modified:   src/components/Main/CardList.js
	modified:   src/components/Main/Main.js
	modified:   src/components/Main/SearchBar.js
	modified:   src/index.js
	modified:   src/pages/Shared/Shared.js
	modified:   src/utils/formatDate.js
	modified:   src/utils/timeAgo.js

jinhok96 committed:
modified:   package-lock.json
	modified:   package.json
	modified:   src/App.js
	modified:   src/components/Common/LoginButton.js
	modified:   src/components/Footer/Footer.js
	modified:   src/components/Header/Gnb.js
	modified:   src/components/Header/SharedHeaderContent.js
	modified:   src/components/Main/Card.js
	modified:   src/components/Main/SearchBar.js
	modified:   src/index.js
	modified:   src/services/api.js
	modified:   src/utils/formatDate.js
	modified:   src/utils/timeAgo.js

jinhok96 committed:
chore: prop-types 추가
	modified:   package-lock.json
	modified:   package.json
	modified:   src/components/Common/Button.js
	modified:   src/services/api.js

jinhok96 committed:
chore: 여러가지 eslint 관련 오류 해결
	modified:   public/index.html
	modified:   src/components/Main/Card.js
	modified:   src/components/Main/CardList.js
	modified:   src/index.js

jinhok96 committed:
chore: 함수 선언 방식을 화살표 함수로 변경
	modified:   src/components/Main/SearchBar.js

jinhok96 committed:
chore: 동일한 로직을 함수로 분리
	modified:   src/services/api.js

jinhok96 committed:
chore: 중복된 기본값 설정 제거
	modified:   src/components/Main/Card.js

jinhok96 committed:
chore: handleChange 함수의 이름을 명시적으로 변경
	modified:   src/components/Main/SearchBar.js

jinhok96 committed:
chore: 함수의 이름을 명시적으로 변경
	modified:   src/components/Main/Card.js

jinhok96 committed:
chore: api 선언부 에러 처리를 throw new Error로 변경
	modified:   src/services/api.js

jinhok96 committed:
chore: 중복된 null처리 제거
	modified:   src/components/Header/Gnb.js
	modified:   src/components/Header/SharedHeaderContent.js

jinhok96 committed:
feat: ErrorMessage 컴포넌트 추가
	new file:   src/components/Common/ErrorMessage.css
	new file:   src/components/Common/ErrorMessage.js
	modified:   src/components/Header/SharedHeaderContent.js
	modified:   src/services/api.js

jinhok96 committed:
chore: 에러 메시지 출력 추가 및 스타일 수정
	modified:   src/components/Header/Gnb.js
	modified:   src/services/api.js

jinhok96 committed:
chore: 에러 메시지 추가
	modified:   src/components/Main/CardList.js

jinhok96 committed:
style: 에러 메시지 스타일 수정
	modified:   src/components/Common/ErrorMessage.css

jinhok96 committed:
chore: 테스트용 잘못된 api 주소 원래대로 복구
	modified:   src/services/api.js

jinhok96 committed:
chore: 헤더 콘텐츠에 children prop 적용
	modified:   src/components/Header/Header.js
	modified:   src/pages/Shared/Shared.js

jinhok96 committed:
chore: 메인 콘텐츠에 children prop 적용
	modified:   src/components/Main/Main.js
	modified:   src/pages/Shared/Shared.js

jinhok96 committed:
refactor: SearchBar에 css module, classnames 적용
	modified:   package-lock.json
	modified:   package.json
	modified:   src/components/Main/SearchBar.js
	renamed:    src/components/Main/SearchBar.css -> src/components/Main/SearchBar.module.css

jinhok96 committed:
refactor: 버튼에 css module, classnames 적용
	modified:   src/components/Common/Button.js
	renamed:    src/components/Common/LoginButton.css -> src/components/Common/Button.module.css
	modified:   src/components/Common/LoginButton.js
	renamed:    src/components/Common/Button.css -> src/components/Common/LoginButton.module.css

jinhok96 committed:
refactor: 에러 메시지에 css module, classnames 적용
	modified:   src/components/Common/ErrorMessage.js
	renamed:    src/components/Common/ErrorMessage.css -> src/components/Common/ErrorMessage.module.css

jinhok96 committed:
refactor: gnb에 css module, classnames 적용
	modified:   src/components/Header/Gnb.js
	renamed:    src/components/Header/Gnb.css -> src/components/Header/Gnb.module.css

jinhok96 committed:
refactor: 헤더에 css module, classnames 적용
	modified:   src/components/Header/Header.js
	renamed:    src/components/Header/Header.css -> src/components/Header/Header.module.css

jinhok96 committed:
refactor: 헤더 콘텐츠에 css module, classnames 적용
	modified:   src/components/Header/SharedHeaderContent.js
	renamed:    src/components/Header/SharedHeaderContent.css -> src/components/Header/SharedHeaderContent.module.css

jinhok96 committed:
refactor: 카드에 css module, classnames 적용
	modified:   src/components/Main/Card.js
	renamed:    src/components/Main/Card.css -> src/components/Main/Card.module.css

jinhok96 committed:
refactor: 카드리스트에 css module, classnames 적용
	modified:   src/components/Main/CardList.js
	renamed:    src/components/Main/CardList.css -> src/components/Main/CardList.module.css

jinhok96 committed:
refactor: 메인에 css module, classnames 적용
	modified:   src/components/Main/Main.js
	renamed:    src/components/Main/Main.css -> src/components/Main/Main.module.css

jinhok96 committed:
chore: 중복된 클래스 제거
	modified:   src/components/Main/SearchBar.js

jinhok96 committed:
chore: 검색바 focus 테두리 제거
	modified:   src/components/Main/SearchBar.module.css

jinhok96 committed:
feat: 라우터 추가
	modified:   package-lock.json
	modified:   package.json
	modified:   src/App.js
	modified:   src/components/Header/Gnb.js

jinhok96 committed:
chore: 로고 클릭 시 페이지 이동할 때 최상단으로 이동
	modified:   src/components/Header/Gnb.js

jinhok96 committed:
refactor: 푸터에 css module, classnames 적용
	modified:   src/components/Footer/Footer.js
	renamed:    src/components/Footer/Footer.css -> src/components/Footer/Footer.module.css

jinhok96 committed:
chore: 소셜 링크 클릭 시 새 창에서 열기
	modified:   src/components/Footer/Footer.js
